File: src/Binary Tree Postorder Traversal.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> postorderTraversal(TreeNode *root) {
        ans.clear();
        doTraversal(root);
        return ans;
    }

    void doTraversal(TreeNode *t) {
        if(!t)
            return;
        if(t->left)
            doTraversal(t->left);
        if(t->right)
            doTraversal(t->right);
        ans.push_back(t->val);
    }
private:
    vector<int> ans;
};

File: python/Binary Tree Postorder Traversal.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return a list of integers
    def postorderTraversal(self, root):
        self.ans = []
        self.doTraversal(root)
        return self.ans

    def doTraversal(self, t):
        if t == None:
            return None
        if t.left != None:
            self.doTraversal(t.left)
        if t.right != None:
            self.doTraversal(t.right)
        self.ans.append(t.val)